File: lib/repo.dart


class Repo {
  final String name;
  final String description;
  final String url;

  Repo({required this.name, required this.description, required this.url});

  factory Repo.fromJson(Map<String, dynamic> json) {
    return Repo(
      name: json['name'],
      description: json['description'] ?? "No description",
      url: json['html_url'],
    );
  }
}

File: src/types/repo.ts
export interface Repo {
    name: string;
    description: string | null;
    html_url: string;
    id: number;
    language?: string;
    stars?: number;
    updated_at?: string;
    homepage?: string | null;
}

export interface FeaturedRepo extends Repo {
    blurb?: string | null;
    highlight?: string | null;
    priority?: number | string | null;
}

export interface TimelineItem {
    year: string;
    title: string;
    description: string;
}
